File: python/1-50/3. Longest Substring Without Repeating Characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        if not s or len(s) <= 1:
            return len(s)
        dict1 = {}
        start, end = 0, 0
        substr = ""
        maxval = 0
        for i in range(len(s)):
            if s[i] in dict1 and dict1[s[i]] >= start:
                start = dict1[s[i]] + 1
            end += 1
            maxval = max(maxval, end - start)
            dict1[s[i]] = i
        return maxval

File: cpp/1-50/3. Longest Substring Without Repeating Characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        vector<int> flags(256, -1);
        if(s.empty())
            return 0;
        int res = 1;
        int start = -1;
        for(int i = 0; i < s.size(); i++){
            if(flags[s[i]] > start){
                start = flags[s[i]];
            }
            res = max(res, i - start);
            flags[s[i]] = i;
        }
        return res;
    }
};